src/Symfony/Component/PasswordHasher/Tests/Hasher/NativePasswordHasherTest.php

```diff
@@ -98,16 +98,44 @@ public function testBcryptWithLongPassword()
         $this->assertTrue($hasher->verify($hasher->hash($plainPassword), $plainPassword));
     }
 
-    public function testBcryptWithNulByte()
+    /**
+     * @requires PHP < 8.4
+     */
+    public function testBcryptWithNulByteWithNativePasswordHash()
     {
         $hasher = new NativePasswordHasher(null, null, 4, \PASSWORD_BCRYPT);
         $plainPassword = "a\0b";
 
-        if (\PHP_VERSION_ID < 80218 || \PHP_VERSION_ID >= 80300 && \PHP_VERSION_ID < 80305) {
-            // password_hash() does not accept passwords containing NUL bytes since PHP 8.2.18 and 8.3.5
-            $this->assertFalse($hasher->verify(password_hash($plainPassword, \PASSWORD_BCRYPT, ['cost' => 4]), $plainPassword));
+        try {
+            $hash = password_hash($plainPassword, \PASSWORD_BCRYPT, ['cost' => 4]);
+        } catch (\Throwable $throwable) {
+            // we skip the test in case the current PHP version does not support NUL bytes in passwords
+            // with bcrypt
+            //
+            // @see https://github.com/php/php-src/commit/11f2568767660ffe92fbc6799800e01203aad73a
+            if (false !== strpos($throwable->getMessage(), 'Bcrypt password must not contain null character')) {
```

Comment:
```suggestion
            if (!str_contains($throwable->getMessage(), 'Bcrypt password must not contain null character')) {
```

Reply:
This function is not available for PHP < 8, but I wish 😄

Reply:
we already require symfony/polyfill-php80 :)

Reply:
Oh that's right, here you go!